File: Python/RemoveDuplicatesFromSortedList.py
class ListNode():
    def __init__(self,val,next) -> None:
        self.val = val
        self.next = next


def removeDuplicates(head):
    slow = fast = head

    while fast and fast.next:
        if fast.next.val != fast.val:
            slow.next = fast.next
            slow = fast.next
        else:
            if fast.next.next == None:
                slow.next = None
        
        fast = fast.next

    return head

File: Javascript/RemoveDuplicatesFromSortedList.js
var ListNode = function(val,next){
    this.val = (val === undefined) ? 0 : val
    this.next = (next === undefined) ? null : next
}

var removeDuplicates = function(head){
    let slow = head
    let fast = head

    while(fast && fast.next){
        if(fast.next.val !== fast.val){
            slow.next = fast.next
            slow = fast.next
        }
        else{
            if(fast.next.next === null){
                slow.next = null
            }
        }
        fast = fast.next
    }

    return head
}